onyo/lib/onyo.py

```diff
@@ -26,12 +27,32 @@ class OnyoProtectedPathError(Exception):
 class Repo:
     """
     """
-    def __init__(self, path: Union[Path, str], init: bool = False) -> None:
+    def __init__(self, path: Union[Path, str], find_root: bool = False,
+                 init: bool = False) -> None:
+        """
+        Instantiates a `Repo` object with `path` as the root directory.
+
+        If `find_root=True` searches the root of a repository from `path`.
+
+        If `init=True`, the `path` will be initialized as a git repo and a
+        `.onyo/` directory will be created.
+        Otherwise the validity of the onyo repository is verified.
+        """
+        path = Path(path)
+
+        if find_root:
+            path = self._find_root(path)
+        else:
+            path = Path(path).resolve()
+
         if init:
             self._init(path)
+        elif not Repo._is_onyo_root(path):
```

Comment:
Why is this calling `Repo._is_onyo_root(path)`, rather than `self._is_onyo_root(path)`? This just implies to me that this method should not be part of the `Repo` class.

Reply:
For the record:
While I do think, that this should eventually go elsewhere, I do think so for a different reason.
Just because it's static doesn't necessarily imply it needs to go. And do find it useful (and even better) to call a static method via the class rather than the instance (`self`). Makes it clear when reading the calling code that there's no dependence on the actual instance w/o needing to lookup the called method.